### ISSUE TITLE
`create_invoice_from_channelmanager` variant with description hash support

### ISSUE THREAD

**Issue by TheBlueMatt** [lightningdevkit/rust-lightning#1280]
I'm not sure there's much use in description hashes in the wild today, except for lnurl/the lnpay stuff, which we'd of course like to support. We really need a variant of `create_invoice_from_channelmanager` which supports a description hash instead of explicit description for that, though, and will probably want something similar for `create_phantom_invoice` after #1199 is merged.

Tagging good first issue, but if someone does take this up, probably base it on #1199 or wait until that is merged.

**Comment by xraid** [comment]
no need for dh in lnurl one can use hash as id for what is described as dh in lnurl spec.

it is just a handle for id in roundtrips

dh is stupid . why have a description if it gets hashed and unreadable at recipient ?

**Comment by xraid** [comment]
a lnurl service could use a hash of the description not necessarily the bolt11 description_hash

**Comment by TheBlueMatt** [comment]
No, the lnurl spec requires you use the description_hash in bolt 11, so to support lnurl you need this.

**Comment by xraid** [comment]
i used hash of preimige in a lnurl service . as it is only used as id in the roundtrip, where the service can id url and hash when incoming

**Comment by xraid** [comment]
You as service can put a teddybear in that fiels as long as You as service can id the teddybear when it comes back, it is not used by the client, only by service ...

**Comment by xraid** [comment]
maybe i am confusing it . lnurl-withdrawal can have teddybears in description_hash field, maybe lnurl-pay can not

**Comment by TheBlueMatt** [comment]
Ah, yes, I was referring to lnurl-pay.
